benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -770,10 +663,155 @@ void GraphicsBenchmarkApp::DrawExtraInfo()
+                if (i == currentDrawCallCount - 1) {
+                    indexCount += (currentSphereCount * mSphereIndexCount - currentDrawCallCount * indicesPerDrawCall);
+                }
+                uint32_t firstIndex = i * indicesPerDrawCall;
+                frame.cmd->DrawIndexed(indexCount, /* instanceCount = */ 1, firstIndex);
+            }
+        }
+    }
+    frame.cmd->EndRenderPass();
+}
+
+void GraphicsBenchmarkApp::RenderFullscreenQuads(grfx::SwapchainPtr swapchain, PerFrame& frame, uint32_t imageIndex)
+{
+    grfx::RenderPassPtr currentRenderPass = GetRenderpass(swapchain, imageIndex);
+
+    if (pFullscreenQuadsCount->GetValue() > 0) {
+        bool singleRenderpass = pFullscreenQuadsSingleRenderpass->GetValue();
+        frame.cmd->BindGraphicsPipeline(mFullscreenQuads.pipeline);
+        frame.cmd->BindVertexBuffers(1, &mFullscreenQuads.vertexBuffer, &mFullscreenQuads.vertexBinding.GetStride());
+
+        if (singleRenderpass) {
+            currentRenderPass = swapchain->GetRenderPass(imageIndex);
+            PPX_ASSERT_MSG(!currentRenderPass.IsNull(), "render pass object is null");
+            frame.cmd->BeginRenderPass(currentRenderPass);
+        }
+
+        for (size_t i = 0; i < pFullscreenQuadsCount->GetValue(); ++i) {
+            if (!singleRenderpass) {
+                currentRenderPass = swapchain->GetRenderPass(imageIndex);
+                PPX_ASSERT_MSG(!currentRenderPass.IsNull(), "render pass object is null");
+                frame.cmd->BeginRenderPass(currentRenderPass);
+            }
+
+            if (pFullscreenQuadsColor->GetIndex() > 0) {
+                float3 colorValues = kFullscreenQuadsColorsValues[pFullscreenQuadsColor->GetIndex()];
+                frame.cmd->PushGraphicsConstants(mFullscreenQuads.pipelineInterface, 3, &colorValues);
+            }
+            else {
+                uint32_t noiseQuadRandomSeed = (uint32_t)i;
+                frame.cmd->PushGraphicsConstants(mFullscreenQuads.pipelineInterface, 1, &noiseQuadRandomSeed);
+            }
+            frame.cmd->Draw(3, 1, 0, 0);
+
+            if (!singleRenderpass) {
+                frame.cmd->EndRenderPass();
+
+                // Force resolve by transitioning image layout
+                frame.cmd->TransitionImageLayout(currentRenderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_RENDER_TARGET, grfx::RESOURCE_STATE_SHADER_RESOURCE);
+                frame.cmd->TransitionImageLayout(currentRenderPass->GetRenderTargetImage(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_SHADER_RESOURCE, grfx::RESOURCE_STATE_RENDER_TARGET);
+            }
+        }
+
+        if (singleRenderpass) {
+            frame.cmd->EndRenderPass();
+        }
+    }
+}
+
+void GraphicsBenchmarkApp::RenderGUI(grfx::SwapchainPtr swapchain, PerFrame& frame, uint32_t imageIndex)
+{
+    grfx::RenderPassPtr currentRenderPass = GetRenderpass(swapchain, imageIndex);
+
+    if (GetSettings()->enableImGui) {
```

Comment:
If you take out the image layout code out of this function you can do :
```
if (!GetSettings()->enableImGui()) {
    return;
}
```

Reply:
Obsolete